### PR TITLE
feat: Add workspace:create command instead of workspace:start

### PR DESCRIPTION
**Pull request by mmorhun**
### What does this PR do?
Adds new `workspace:create` command to `chectl`, which is supposed to replace `worspace:start` which become deprecated.

`workspace:create` command has `--start` option which starts the workspace right after creation.

In case of using `workspace:start` command, deprecation warning will be shown:
output

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/14344

### CODE REVIEW

**Comment by tolusha** [comment]
1. Remove flag from both commands:
`'listr-renderer': listrRenderer`
and use 
`{ renderer: 'silent' }`
2. Use `cli.warn`  or `cli.log` to print information

**Comment by tolusha** [comment]
3. `yarn pack` to update `README.md`

File: yarn.lock [hunk 1]
```diff
@@ -123,6 +123,10 @@
     exec-sh "^0.3.2"
     minimist "^1.2.0"
 
+"@eclipse-che/api@latest":
+  version "7.5.0-SNAPSHOT"
```
Comment (tolusha) [inline yarn.lock]: looks strange

File: src/commands/workspace/start.ts [hunk 2]
```diff
@@ -8,72 +8,29 @@
...
-    'listr-renderer': listrRenderer
-  }
-
-  async checkToken(flags: any, ctx: any) {
-    if (ctx.isAuthEnabled && !flags['access-token']) {
-      this.error('E_AUTH_REQUIRED - Eclipse Che authentication is enabled and an access token need to be provided (flag --access-token).')
-    }
-  }
+export default class Start extends Create {
+  static description = 'Creates and starts workspace from devfile'
```
Comment (tolusha) [inline src/commands/workspace/start.ts]: I believe that this command is still used.
So, let's keep the current functionality + add warning...

File: src/commands/workspace/create.ts [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+    help: flags.help({ char: 'h' }),
+    chenamespace: cheNamespace,
+    devfile: string({
+      char: 'f',
+      description: 'path or URL to a valid devfile',
+      env: 'DEVFILE_PATH',
+      required: true,
+    }),
+    name: string({
+      description: 'workspace name: overrides the workspace name to use instead of the one defined in the devfile. Works only for devfile',
```
Comment (tolusha) [inline src/commands/workspace/create.ts]: The devfile is the only way. The description should be adjusted

File: src/commands/workspace/create.ts [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+  static description = 'Creates a workspace from devfile'
+
+  static flags = {
+    help: flags.help({ char: 'h' }),
+    chenamespace: cheNamespace,
+    devfile: string({
+      char: 'f',
+      description: 'path or URL to a valid devfile',
+      env: 'DEVFILE_PATH',
+      required: true,
```
Comment (tolusha) [inline src/commands/workspace/create.ts]: We can make it optional. If flag isn't set then let's use the devfile from the current directory

File: src/commands/workspace/create.ts [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+import { Command, flags } from '@oclif/command'
+import { boolean, string } from '@oclif/parser/lib/flags'
+import { cli } from 'cli-ux'
+import * as Listr from 'listr'
+import * as notifier from 'node-notifier'
+
+import { CheHelper } from '../../api/che'
+import { accessToken, cheNamespace, listrRenderer } from '../../common-flags'
+export default class Create extends Command {
+  static description = 'Creates a workspace from devfile'
```
Comment (tolusha) [inline src/commands/workspace/create.ts]: `from a devfile`

File: yarn.lock (hunk 1 above)
Comment (AndrienkoAleksandr) [reply yarn.lock]: Hmm. It is really latest version... https://www.npmjs.com/package/@eclipse-che/api

Comment (mmorhun) [reply yarn.lock]: Nothing strange. I just used `latest` in package json, so it was resolved to `7.5.0-SNAPSHOT`

Comment (tolusha) [reply yarn.lock]: Why do we need this at all?

Comment (AndrienkoAleksandr) [reply yarn.lock]: I think @mmorhun wants to use dts types for types

Comment (mmorhun) [reply yarn.lock]: Yes, I think it is good to have response structure typings generated from Che servers DTOs

Comment (tolusha) [reply yarn.lock]: But is it outdated version?

Comment (AndrienkoAleksandr) [reply yarn.lock]: @azatsarynnyy @vparfonov  Do we have newer version of the @eclipse-che/api or release mechanizm become broken after moving to the centos ci?

Comment (azatsarynnyy) [reply yarn.lock]: @AndrienkoAleksandr do you know where's the repo with the sources of `@eclipse-che/api`?

Comment (AndrienkoAleksandr) [reply yarn.lock]: I think publish here https://github.com/eclipse/che/tree/master/typescript-dto

Comment (azatsarynnyy) [reply yarn.lock]: Sorry @AndrienkoAleksandr but I have no idea who is responsible now for `@eclipse-che/api`. Try to ask in Eclipse Che MM channel tomorrow.

File: src/commands/workspace/create.ts (hunk 3 above)
Comment (mmorhun) [reply src/commands/workspace/create.ts]: Are you suggesting to remove this option at all?

Comment (tolusha) [reply src/commands/workspace/create.ts]: Just remove that
`Works only for devfile`

File: yarn.lock (hunk 1 above)
Comment (mmorhun) [reply yarn.lock]: I think it should a part of release process.

Comment (tolusha) [reply yarn.lock]: Let's not use outdated version.

File: src/commands/workspace/start.ts (hunk 2 above)
Comment (mmorhun) [reply src/commands/workspace/start.ts]: That's exactly what I do.

File: yarn.lock (hunk 1 above)
Comment (mmorhun) [reply yarn.lock]: I don't think there were some major changes. But we need to raise this question.

File: src/commands/workspace/create.ts (hunk 4 above)
Comment (mmorhun) [reply src/commands/workspace/create.ts]: ok

File: yarn.lock (hunk 1 above)
Comment (mmorhun) [reply yarn.lock]: I've create issue for this problem: https://github.com/eclipse/che/issues/16436

File: src/api/che.ts [hunk 6]
```diff
@@ -197,13 +199,37 @@ export class CheHelper {
...
   }
 
-  async createWorkspaceFromDevfile(namespace: string | undefined, devfilePath = '', workspaceName: string | undefined, accessToken = ''): Promise<string> {
+  async startWorkspace(cheNamespace: string, workspaceId: string, accessToken?: string): Promise<void> {
+    const cheUrl = await this.cheURL(cheNamespace)
+    const endpoint = `${cheUrl}/api/workspace/${workspaceId}/runtime`
+    let response
+
+    const headers: {[key: string]: string} = {}
+    if (accessToken && accessToken.length > 0) {
```
Comment (tolusha) [inline src/api/che.ts]: `if (accessToken) ` enough?

File: src/api/che.ts [hunk 7]
```diff
@@ -197,13 +199,37 @@ export class CheHelper {
...
+      headers.Authorization = accessToken
+    }
+    try {
+      response = await this.axios.post(endpoint, undefined, { headers })
+    } catch (error) {
+      if (error.response && error.response.status === 404) {
```
Comment (tolusha) [inline src/api/che.ts]: `getCheApiError` function actually catches  404 error (rebase on master pls)

File: src/commands/workspace/create.ts [hunk 8]
```diff
@@ -0,0 +1,122 @@
...
+          if (ctx.isAuthEnabled && !flags['access-token']) {
+            this.error('E_AUTH_REQUIRED - Eclipse Che authentication is enabled and an access token needs to be provided (flag --access-token).')
+          }
+
+          let devfilePath: string
+          if (flags.devfile) {
+            devfilePath = flags.devfile
+          } else if (fs.existsSync('devfile.yaml')) {
+            devfilePath = 'devfile.yaml'
+          } else if (fs.existsSync('devfile.yml')) {
```
Comment (tolusha) [inline src/commands/workspace/create.ts]: duplication ?

Comment (mmorhun) [reply src/commands/workspace/create.ts]: No, the extension is different

File: src/api/che.ts (hunk 6 above)
Comment (mmorhun) [reply src/api/che.ts]: yes

File: src/api/che.ts (hunk 7 above)
Comment (mmorhun) [reply src/api/che.ts]: Not only...

Comment (mmorhun) [reply src/api/che.ts]: I want to handle 404 error in another way for this specific request.

**Comment by mmorhun** [comment]
Rebased

**Pull request merged**